polyphemus/lib/etls/add_watch_folder_base_etl.rb

```diff
@@ -0,0 +1,82 @@
+require_relative "../metis_folder_etl"
+
+class Polyphemus::AddWatchFolderBaseEtl < Polyphemus::MetisFolderEtl
+  def initialize(project_bucket_pairs:, model_name:, folder_name_globs: [], limit: 20)
+    @model_name = model_name
+    super(
+      project_bucket_pairs: project_bucket_pairs,
+      folder_name_globs: folder_name_globs,
+      limit: limit,
+    )
+  end
+
+  def process(cursor, folders)
+    # All of these folders should contain processed files that
+    #   we want to watch and link in when / if they change.
+    logger.info("Found #{folders.length} updated folders: #{folders.map { |f| f.folder_path }.join(",")}")
+
+    new_folders_to_watch = new_folders(cursor, folders)
+
+    logger.info("Found #{new_folders_to_watch.length} folders to watch: #{new_folders_to_watch.map { |f| f.folder_path }.join(",")}")
+
+    create_records(cursor, new_folders_to_watch)
```

Comment:
So just a note as I work on this ... I think this may not work as intended... because there isn't an explicit regex that filters folders within the returned set (every folder returned from the Metis `find` call is added as a watch folder), we actually won't catch folders that no longer match the globs -- since they don't match the globs. And hence not sure we can remove them, without scanning for all folders on Metis...

Reply:
Ahhh, see I was thinking this process would work differently -- it would follow changes to all folders (process gets called for all folder changes, not just those that match the glob), and that the process function itself would then do the matching -- any folder that matches the glob (etl ruby side, not api side) would have watch folder records created, and those that didn't would have any existing watch folder by their id removed.

Does that make some sense?

Reply:
Hm, okay, we should talk about this today, perhaps, but I still don't think the removal will work as intended. The issue is that subdirectories aren't changed when a parent directory changes, but the parent directory changes might result in the regex not matching anymore. So I think we may never know when a folder should lose its watch status, directly.

For example:

Let's say we're watching for sub-directories of a folder named `output/`. So the regex is something like `/output\/.*$/`. In this case, a folder with a folder_path like `output/PATIENT001.T1.rna.live` would match and be added as a watch folder. Call `output/` id 1 and `output/PATIENT001.T1.rna.live` as id 2.

Now, let's say someone changes `output/` to `wrong_output/`. The only Metis folder that is going to come through the cursor is id 1, and not id 2. Note that id 1 is not a watch folder. How do we know to remove id 2 as a watch folder? We could try to scan all watch folders with `folder_id=1`, but that only works one level deep, since we're not duplicating the entire Metis folder structure, and we could miss some changes that would affect path regex matching from parent folders well up the tree.